internal/clients/http/image_builder/image_client.go

```diff
@@ -45,21 +44,20 @@ func (c *ibClient) Ready(ctx context.Context) error {
 	defer span.End()
 
 	logger := logger(ctx)
-	resp, err := c.client.GetReadiness(ctx, headers.AddImageBuilderIdentityHeader, headers.AddEdgeRequestIdHeader)
+	resp, err := c.client.GetReadinessWithResponse(ctx, headers.AddImageBuilderIdentityHeader, headers.AddEdgeRequestIdHeader)
 	if err != nil {
 		logger.Error().Err(err).Msg("Readiness request failed for image builder")
 		return err
 	}
-	defer func() {
-		if tempErr := resp.Body.Close(); tempErr != nil {
-			logger.Error().Err(tempErr).Msg("Readiness request for image builder: response body close error")
-		}
-	}()
 
-	err = http.HandleHTTPResponses(ctx, resp.StatusCode)
-	if err != nil {
-		return fmt.Errorf("ready call: %w", err)
+	if resp == nil {
+		return fmt.Errorf("ready call: empty response: %w", clients.UnexpectedBackendResponse)
 	}
+
+	if resp.StatusCode() < 200 || resp.StatusCode() > 299 {
+		return fmt.Errorf("ready call: %w: %d", clients.UnexpectedBackendResponse, resp.StatusCode())
+	}
```

Comment:
I think you can now use `resp.JSON200 == nil` (in all the three functions) to keep the consistency and prevent copy and paste error.

Reply:
Yeah, that is possible, but we do not access resp.JSON200 structure, so we technically do not need this to be only 200, right? Or do we want to error out on things like 201 etc., in case of readiness call?